### PR TITLE
Setup Documentation, GitHub Actions workflows, linting, and styling standards for NM actions

### PR DESCRIPTION
**Pull request by markurtz**
### Summary:

This pull request sets up the foundational components for the repository, including GitHub Actions workflows for development and main branches, documentation updates, linting, and style standards. The changes introduce a `.gitignore` file, GitHub workflows (`development.yml` and `main.yml`), and configuration files for `ruff`, `mypy`, `yamlfix`, and markdown formatting, ensuring code quality, style consistency, and efficient CI/CD pipelines.

### Details:

- Added GitHub Actions workflows for development and main branches.
- Configured `tox` environments to run quality checks and style enforcement.
- Introduced `.gitignore` to filter unnecessary files.
- Configured `ruff`, `mypy`, `yamlfix`, and markdown formatting settings to enforce quality and style standards across the repository.
- Updated README with instructions on setting up GitHub Actions and contribution guidelines.
- Added a pull request template for consistency in contributions.
- Configured automatic formatting and linting for YAML and markdown files using `yamlfix` and `mdformat`.

### Test Plan:

- Ensure the following GitHub Actions workflows execute successfully:
    - `development.yml` on all branches for code quality checks.
    - `main.yml` on `main` branch for production-quality validation.
- Run `tox -e quality` to confirm that all files meet quality standards.
- Verify `tox -e style` applies formatting and styling corrections without errors.

### Related Issues:

N/A

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
-"please provide a brief summary"
+## Summary:
 
-TEST PLAN:
-"please outline how the changes were tested"
+Short paragraph detailing the pull request changes and reasoning in addition to any relevant context.
 
+## Details:
+
+- Detailed list of changes made in the pull request
```
Comment (andy-neuma) [inline .github/pull_request_template.md:7]: feel summary is enough, no need to have this additional list.

File: .github/workflows/development.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+name: Development Branch Workflow
+
+on:
+  pull_request:
```
Comment (andy-neuma) [inline .github/workflows/development.yml]: we shouldn't be triggering on "pull_request" ... it should just be "on push"

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,162 @@
+# Byte-compiled / optimized / DLL files
```
Comment (andy-neuma) [inline .gitignore:1]: this looks like it is copied from one of our other REPOs. there isn't really a need for this.

**Review by andy-neuma — DISMISSED** [review]
thanks

File: .github/workflows/development.yml
Comment (dhuangnm) [inline .github/workflows/development.yml:1]: Can we just combine development.yml and main.yml to check for every branch?

File: .gitignore (hunk 3 above)
Comment (dhuangnm) [inline .gitignore:1]: Can we also add *.swp?

**Review by derekk-nm — DISMISSED** [review]
thanks for closing the loop on this Andy

File: actions/pypi_build/action.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
+---
```
Comment (dhuangnm) [inline actions/pypi_build/action.yaml]: This file is already removed from latest main. Can you please rebase from latest main before merging?

File: actions/testmo-run-create/action.yml [hunk 5]
```diff
@@ -1,27 +1,25 @@
...
     required: true
   testmo_token:
-    description: 'testmo token'
+    description: testmo token
     required: true
   source:
-    description: "source for testmo, e.g. 'build-test'"
+    description: source for testmo, e.g. 'build-test'
     required: true
-  project_id:
```
Comment (dhuangnm) [inline actions/testmo-run-create/action.yml:13]: project_id is needed. Please rebase from latest main.

**Review by dhuangnm — CHANGES_REQUESTED** [review]

File: .github/workflows/development.yml
Comment (andy-neuma) [reply .github/workflows/development.yml:1]: yeah

File: .gitignore (hunk 3 above)
Comment (andy-neuma) [reply .gitignore:1]: sure, but this file is complete overkill. this REPO should be very light weight. :)

File: actions/testmo-run-create/action.yml (hunk 5 above)
Comment (andy-neuma) [reply actions/testmo-run-create/action.yml:13]: okay, looks like a bad merge on my part ... :)

File: actions/pypi_build/action.yaml (hunk 4 above)
Comment (andy-neuma) [reply actions/pypi_build/action.yaml]: yeah, it showed locally as deleted when i updated.

**Review by dhuangnm — APPROVED** [review]
Looks good, thanks!

**Pull request merged**